Adafruit_MQTT.cpp

```diff
@@ -653,7 +653,7 @@ uint16_t Adafruit_MQTT::publishPacket(uint8_t *packet, const char *topic,
   len += bLen; // payload length
 
   // Now you can start generating the packet!
-  p[0] = MQTT_CTRL_PUBLISH << 4 | qos << 1;
+  p[0] = MQTT_CTRL_PUBLISH << 4 | qos << 1 | retain;
```

Comment:
hmm... I would think it would be safer if you did not expect the compiler to map false to 0 and true to 1.
I can see that retain is as simple as setting bit 0, based on this spec:
https://docs.solace.com/MQTT-311-Prtl-Conformance-Spec/MQTT%20Control%20Packet%20format.htm

But mixing types is kinda scary. How about rewriting it to something like:
```
p[0] = MQTT_CTRL_PUBLISH << 4 | qos << 1 | (retain ? 1 : 0);
```

Reply:
Makes sense, changed it.